src/metabase/search/filter.clj

```diff
@@ -62,26 +62,37 @@
      [:= (search.config/column-with-model-alias model :active) true]
      [:= (search.config/column-with-model-alias model :visibility_type) nil]]))
 
-(defn- search-string-clause
-  [model query searchable-columns]
-  (when query
-    (into [:or]
-          (for [column searchable-columns
-                token (search.util/tokenize (search.util/normalize query))]
-            (if (and (= model "indexed-entity") (premium-features/sandboxed-or-impersonated-user?))
-              [:= 0 1]
-
-              [:like
-               [:lower column]
-               (search.util/wildcard-match token)])))))
-
 (mu/defn ^:private search-string-clause-for-model
-  [model          :- SearchableModel
-   search-context :- SearchContext]
-  (search-string-clause model (:search-string search-context)
-                        (map #(search.config/column-with-model-alias model %)
-                             (search.config/searchable-columns-for-model model))))
-
+  [model                :- SearchableModel
+   search-context       :- SearchContext
+   search-native-query? :- [:maybe :boolean]]
+  (when-let [query (:search-string search-context)]
+    (into
+     [:or]
+     (for [column (cond->> (search.config/searchable-columns-for-model model)
+                    (not search-native-query?)
+                    (remove #{:dataset_query})
+
+                    true
+                    (map #(search.config/column-with-model-alias model %)))
+           token (->> (search.util/normalize query)
+                      search.util/tokenize
+                      (map search.util/wildcard-match))]
```

Comment:
how about I rename this to `wildcarded-token`? `token` in this context is only used within `like` clause so I want to avoid manually wildcard it in all the places we use it.

Reply:
`wildcarded-token` sounds good